From dfb4333dc655f48a0467a6472d8566d3ff8fbc64 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Wed, 8 Feb 2023 10:24:58 +1100
Subject: [PATCH 2/2] set CMAKE_CXX_STANDARD in mlir/CMakeLists.txt

---
 mlir/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mlir/CMakeLists.txt b/mlir/CMakeLists.txt
index f00c819b25..4594b8830e 100644
--- a/mlir/CMakeLists.txt
+++ b/mlir/CMakeLists.txt
@@ -22,6 +22,7 @@ endif()
 
 # Must go below project(..)
 include(GNUInstallDirs)
+set(CMAKE_CXX_STANDARD 17)
 
 if(MLIR_STANDALONE_BUILD)
   find_package(LLVM CONFIG REQUIRED)
-- 
2.38.1.windows.1

